samples/bot-configuration-app/nodejs/bots/teamsBot.js

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler, CardFactory, MessageFactory } = require("botbuilder");
+var chosenFlow = "";
+class TeamsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+        this.onMembersAdded(async (context, next) => {
+            const membersAdded = context.activity.membersAdded;
+            for (let member = 0; member < membersAdded.length; member++) {
+                if (membersAdded[member].id !== context.activity.recipient.id) {
+                    await context.sendActivity("Hello and welcome! With this sample you can see the functionality of static and dynamic search in adaptive card");
+                }
+            }
+
+            await next();
+        });
+
+        this.onMessage(async (context, next) => {
+
+            if (context.activity.text != null) {
+                if (context.activity.text.toLowerCase().trim() === "chosen flow" || context.activity.text.toLowerCase().trim() === "<at>typeahead search adaptive card</at> chosen flow") {
+                    const replyActivity = MessageFactory.text(`Bot configured for ${chosenFlow} flow`);
+                    await context.sendActivity(replyActivity);
+                }
+            }
+            else if (context.activity.value != null) {
+                await context.sendActivity("Selected option is: " + context.activity.value.choiceselect);
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    }
+
+    async onInvokeActivity(context) {
+
+        if (context._activity.name == "config/fetch") {
+            const adaptiveCard = CardFactory.adaptiveCard(this.adaptiveCardForDynamicSearch());
+            try {
+                return {
+                    status: 200,
+                    body: {
+                        config: {
+                            type: 'continue',
+                            value: {
+                                card: adaptiveCard,
+                                height: 400,
+                                title: 'Task module fetch response',
+                                width: 300
+                            }
+                        }
+                    }
+                }
+            } catch (e) {
+                console.log(e);
+            }
+        }
+
+        if (context._activity.name == "config/submit") {
```

Comment:
What is this condition?

Reply:
This condition arises from the additional functionality introduced in the SDK for bot reconfiguration. Please refer to the sentence below.

Upon installation in either team or groupChat scope, the fetchTask initiates a config/fetch. The bot responds with an adaptive card. On submitting details, a config/submit is returned to the bot, which handles the response appropriately.

Reply:
@Jegadeesh-MSFT - Add this comment/summary to if condition.